### PR TITLE
Fix LRU cache bloat

### PR DESCRIPTION
**Pull request by richardms**
See #135.

### CODE REVIEW

File: lib/server.js [hunk 1]
```diff
@@ -323,10 +334,17 @@ CoAPServer.prototype._handle = function(packet, rsinfo) {
...
     buf.sender = sender
 
+    var key = toKey(rsinfo.address, rsinfo.port,
+        packet, packet.ack || !packet.confirmable)
+    lru.set(key, buf)
+
+    // Start timer to actively prune from LRU cache after expiration
+    buf.lru_timer = setTimeout(function () {
+      lru.get(key);
+    }, that._lruMaxAge + 1000)
```
Comment (mcollina) [inline lib/server.js]: you can create an interval and call `unref()`, then you won't need to reset the timer every time.

File: lib/server.js [hunk 2]
```diff
@@ -83,10 +83,11 @@ function CoAPServer(options, listener) {
   // 32 MB / 1280 = 26214
   // The max lifetime is roughly 200s per packet.
   // Which gave us 131 packets/second guarantee
+  this._lruMaxAge = (parameters.exchangeLifetime * 1000)
   this._lru = LRU({
       max: options.cacheSize || (32768 * 1024)
-    , length: function(n) { return n.length }
-    , maxAge: parameters.exchangeLifetime
+    , length: function(n) { return n.buffer.byteLength }
+    , maxAge: this._lruMaxAge
```
Comment (mcollina) [inline lib/server.js]: there is no need to make `_lruMaxAge` a property, assign it directly here.

**Comment by mcollina** [comment]
Also, do you have some ideas into adding a unit test for this?

File: lib/server.js (hunk 2 above)
Comment (richardms) [inline lib/server.js]: OK. I'll switch to an Interval called once every maxAge that runs a prune() across the whole lru.

File: lib/server.js [hunk 3]
```diff
@@ -237,6 +237,13 @@ CoAPServer.prototype.listen = function(port, address, done) {
   this._port = port
   this._address = address
 
+  if (parameters.pruneTimerPeriod) {
+    // Start LRU pruning timer
+    this._lru.pruneTimer = setInterval(function () {
+      that._lru.prune()
+    }, parameters.pruneTimerPeriod*1000)
```
Comment (mcollina) [inline lib/server.js]: can you please `unref()`  this?

**Comment by richardms** [comment]
Please ignore the PR for the time being - I'm having trouble getting the tests to run reliably on both Windows & Linux

**Comment by richardms** [comment]
OK. Working well on Linux, although Travis CI is failing for node 0.12. Do you think that's something I've done or due to a changing dependency?

**Comment by mcollina** [comment]
I'm ok in dropping support for node v0.12.

Would you like to help maintaining/developing this module?

**Comment by richardms** [comment]
Sorry to disappear - I went off sick for the second half of last week, and I've been catching up ever since.

Time is short for me at the moment, but I'm happy to help the maintenance effort when I can.

File: lib/server.js (hunk 3 above)
Comment (mcollina) [inline lib/server.js]: this is still not `unref()`.

Comment (richardms) [inline lib/server.js]: it is below - one of the mock timer libraries doesn't support `unref()` so I had to put a guard around it.

**Pull request merged**